gammapy/data/counts.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import print_function, division
+from gammapy.spectrum import EnergyBinning
+
+
+__all__ = ['CountsSpectrumDataset',
+           ]
+
+
+class CountsSpectrumDataset(object):
```

Comment:
I started adding `Dataset` when I implemented [EventList](https://gammapy.readthedocs.org/en/latest/api/gammapy.data.EventList.html) and [EventListDataset](https://gammapy.readthedocs.org/en/latest/api/gammapy.data.EventListDataset.html).
There it might make sense, because the event list FITS files contain an extra HDU, the GTI extension.

But for something like this maybe it's simplest to just call it `CountsSpectrum`?
Or we add `Dataset` to all classes in Gammapy that represent datasets?
I don't have a clear concept yet, so for now: @kingj90 – you decide.